core/templates/domain/collection/collection-rights-backend-api.service.ts

```diff
@@ -16,143 +16,168 @@
-        });
+  }
 
-      var putParams = {
-        version: collectionVersion
-      };
-      var requestUrl = (
-        isPublic ? collectionPublishUrl : collectionUnpublishUrl);
+  private _setCollectionStatus(collectionId: string,
+      collectionVersion: number,
+      isPublic: boolean,
+      successCallback: Function,
+      errorCallback: Function): void {
+    let collectionPublishUrl = this.urlInterpolationService
+      .interpolateUrl('/collection_editor_handler/publish/<collection_id>', {
+        collection_id: collectionId
+      });
 
-      $http.put(requestUrl, putParams).then(function(response) {
-        collectionRightsCache[collectionId] = response.data;
-        if (successCallback) {
-          successCallback(response.data);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+    let collectionUnpublishUrl = this.urlInterpolationService
+      .interpolateUrl('/collection_editor_handler/unpublish/<collection_id>', {
+        collection_id: collectionId
       });
-    };
 
-    var _isCached = function(collectionId) {
-      return collectionRightsCache.hasOwnProperty(collectionId);
+    let putParams = {
+      version: collectionVersion
     };
 
-    return {
-      /**
-       * Gets a collection's rights, given its ID.
-       */
-      fetchCollectionRights: function(collectionId) {
-        return $q(function(resolve, reject) {
-          _fetchCollectionRights(collectionId, resolve, reject);
-        });
-      },
+    let requestUrl = (
+      isPublic ? collectionPublishUrl : collectionUnpublishUrl);
 
-      /**
-       * Behaves exactly as fetchCollectionRights (including callback
-       * behavior and returning a promise object), except this function will
-       * attempt to see whether the given collection rights has been
-       * cached. If it has not yet been cached, it will fetch the collection
-       * rights from the backend. If it successfully retrieves the collection
-       * rights from the backend, it will store it in the cache to avoid
-       * requests from the backend in further function calls.
-       */
-      loadCollectionRights: function(collectionId) {
-        return $q(function(resolve, reject) {
-          if (_isCached(collectionId)) {
-            if (resolve) {
-              resolve(collectionRightsCache[collectionId]);
-            }
-          } else {
-            _fetchCollectionRights(collectionId, function(collectionRights) {
-              // Save the fetched collection rights to avoid future fetches.
-              collectionRightsCache[collectionId] = collectionRights;
-              if (resolve) {
-                resolve(collectionRightsCache[collectionId]);
-              }
-            }, reject);
-          }
-        });
-      },
+    this.http.put(requestUrl, putParams).toPromise().then((response: any) => {
+      this.collectionRightsCache[collectionId] = response.data;
```

Comment:
It should be just `response` right, not `response.data`?

Ditto below.

Reply:
Yep, I've overseen this as well as with the `error.data`. I'll get this fixed right away 👍